Client/Frontend/Browser/TabManager.swift

```diff
@@ -55,18 +55,23 @@ class TabManager : NSObject {
     private var configuration: WKWebViewConfiguration
     let storage: RemoteClientsAndTabs?
 
-    init(defaultNewTabRequest: NSURLRequest, storage: RemoteClientsAndTabs? = nil) {
+    private let prefs: Prefs
+
+    init(defaultNewTabRequest: NSURLRequest, storage: RemoteClientsAndTabs? = nil, prefs: Prefs) {
         // Create a common webview configuration with a shared process pool.
         configuration = WKWebViewConfiguration()
         configuration.processPool = WKProcessPool()
+        configuration.preferences.javaScriptCanOpenWindowsAutomatically = !(prefs.boolForKey("blockPopups") ?? true)
 
         self.defaultNewTabRequest = defaultNewTabRequest
         self.storage = storage
         self.navDelegate = TabManagerNavDelegate()
-
+        self.prefs = prefs
         super.init()
 
         addNavigationDelegate(self)
+
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: "prefsDidChange", name: NSUserDefaultsDidChangeNotification, object: nil)
```

Comment:
I can't find the code that sends the `prefsDidChange` notification. Does that need to be added to the `Prefs` code?